### ISSUE TITLE
[git] Replace `git-gutter*` with `diff-hl`? (It was "⛔ Emergency (emacsql): Uninstall the `git-commit' package" makes magit unusable)

### ISSUE THREAD

**Issue by allentiak** [syl20bnr/spacemacs#16555]
#### Description :octocat:
It seems that `git-commit` is no longer needed, since it was recently merged into `magit`.
However, since it is also marked as a dependency of `git-gutter+`, it cannot be manually removed.

This bug makes `magit` completely unusable.

#### Reproduction guide :beetle:
- Start Emacs
- Open any git project
- Do a `magit-status` (`[SPC g s]`)

*Observed behaviour:* :eyes: :broken_heart:

Instead of the `magit-status` buffer, you will see the following one:

```
Please uninstalling `git-commit' and then restart Emacs.
⛔ Emergency (emacsql): Uninstall the `git-commit' package.

The "git-commit.el" library is no longer distributed as a separate
package.  It is now distributed as part of the `magit' package.

The `git-commit' package now does nothing but display this warning.
If it is located earlier on the `load-path' than `magit' is, then that
prevents the proper "git-commit.el" libraries from `magit' from being
loaded.

Please uninstalling `git-commit' and then restart Emacs.
```

When trying to manually uninstall `git-commit`, I get the following error:
```
helm-M-x-execute-command: Package ‘git-commit-20240831.2335’ is used by ‘git-gutter+’ as dependency, not deleting
```

* I have also checked the `git` layer source code, and there is no reference to either `git-commit` nor `git-gutter+`.

* I have also checked the source code for the following Emacs packages, and found no reference to `git-commit`:
  * https://github.com/emacsorphanage/git-gutter
  * https://github.com/emacsorphanage/git-gutter-fringe
  * https://github.com/nonsequitur/git-gutter-plus
  * https://github.com/nonsequitur/git-gutter-fringe-plus

*Expected behaviour:* :heart: :smile:

magit-status should work normally.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 29.4
- Spacemacs: 0.999.0
- Spacemacs branch: develop (rev. e8c461247)
- Graphic display: t
- Running in daemon: nil
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
((auto-completion :variables auto-completion-enable-help-tooltip t auto-completion-enable-snippets-in-popup t auto-completion-enable-sort-by-usage t)
 (clojure :variables cider-overlays-use-font-lock t cider-preferred-build-tool 'clojure-cli cider-repl-buffer-size-limit 250 cider-result-overlay-position 'at-point clojure-backend 'lsp clojure-enable-sayid t clojure-enable-clj-refactor t clojure-enable-kaocha-runner t clojure-enable-linters
          '(clj-kondo joker)
          clojure-indent-style 'align-arguments clojure-align-forms-automatically t clojure-toplevel-inside-comment-form t)
 colors command-log dap emacs-lisp
 (git :variables git-magit-status-fullscreen t magit-diff-adjust-tab-width t magit-diff-paint-whitespace t magit-diff-highlight-trailing t magit-diff-refine-hunk 'all)
 graphql helm html
 (java :variables java-backend 'lsp)
 (javascript :variables javascript-backend 'lsp javascript-fmt-on-save t javascript-fmt-tool 'web-beautify javascript-import-tool 'import-js javascript-repl `nodejs js-indent-level 2 js2-basic-offset 2 js2-mode-show-parse-errors nil js2-mode-show-strict-warnings nil)
 (keyboard-layout :variables kl-layout 'colemak-neio-literal)
 latex lsp
 (markdown :variables markdown-live-preview-engine 'vmd)
 multiple-cursors org parinfer
 (plantuml :variables plantuml-jar-path "/usr/share/plantuml/plantuml.jar" org-plantuml-jar-path "/usr/share/plantuml/plantuml.jar")
 python
 (rust :variables lsp-rust-analyzer-cargo-auto-reload t rustic-format-on-save t)
 (shell :variables shell-default-height 30 shell-default-position 'bottom)
 spacemacs-visual spell-checking sql syntax-checking
 (treemacs :variables treemacs-indentation 1 treemacs-use-filewatch-mode t treemacs-use-follow-mode t treemacs-use-git-mode 'deferred)
 (typescript :variables typescript-backend 'lsp typescript-lsp-linter nil)
 (version-control :variables version-control-diff-tool 'git-gutter+ version-control-global-margin t)
 yaml)
```
- System configuration features: ACL CAIRO DBUS FREETYPE GIF GLIB GMP GNUTLS GPM GSETTINGS HARFBUZZ JPEG JSON LCMS2 LIBOTF LIBSELINUX LIBSYSTEMD LIBXML2 M17N_FLT MODULES NATIVE_COMP NOTIFY INOTIFY PDUMPER PNG RSVG SECCOMP SOUND SQLITE3 THREADS TIFF TOOLKIT_SCROLL_BARS TREE_SITTER WEBP X11 XDBE XIM XINPUT2 XPM GTK3 ZLIB

**Comment by fnussbaum** [comment]
This is due to https://github.com/nonsequitur/git-gutter-plus depending on git-commit, which should be changed to magit. This should be reported upstream, though it looks unmaintained. Note that you can delete the package with `C-u M-x package-delete`, but it will be reinstalled on the next startup. One workaround would be to change the value of `version-control-diff-tool`.

**Comment by smile13241324** [comment]
As the offending package is clearly unmaintained, no comit in the last year, I think we ought to remove it from the layer.

In version control you can define which diff tool you want to use please change it to a different one to avoid the dependency on git gutter+ I am going to remove git gutter+ tomorrow

**Comment by Falis-EPG** [comment]
❤️

**Comment by allentiak** [comment]
Actually, some of the packages I mentioned in the original report are
currently being actively maintained. (Both "no-plus" packages.)

What about switching to those?

More specifically, I mean these two:
https://github.com/emacsorphanage/git-gutter
https://github.com/emacsorphanage/git-gutter-fringe

**Comment by allentiak** [comment]
@smile13241324 
Thanks for answering this so fast.

My actual question would be:
Is `git-gutter+`/`git-gutter-fringe+`'s functionality already part of `magit`'s?
If so... Is there any replacement?
As I said, I could not find any reference to the removed `git-commit` package in any of the four source code trees I mentioned earlier...

**Comment by allentiak** [comment]
Possibly related ticket:
https://github.com/magit/magit/issues/5217#issuecomment-2332729428

**Comment by allentiak** [comment]
@smile13241324 
Maybe a better option is replacing all the `git-gutter*` packages with `diff-hl`?
Context: see https://github.com/magit/magit/issues/5217#issuecomment-2332867507

**Comment by tarsius** [comment]
Hm, it appears I grepped wrong when looking for `git-gutter` in Spacemacs.

> Maybe a better option is replacing all the `git-gutter*` packages with `diff-hl`?

I strongly suggest that `git-gutter+` and `git-gutter-fringe+` are removed completely. These packages were last touched a decade ago and a quick look at the code and issue tracker indicates that they broke a long time ago.

As a side-note, back when the `+` forks were created, I tried to coax the forkee to instead try to get their changes merged upstream. This "schism" was very unfortunate from the start.

I don't know whether the original variants should still be used. They are still being maintained at least, even if it is on the Emacsorphanage.

**Comment by smile13241324** [comment]
I have removed all traces of git-gutter+ and git-gutter-fringe+ from Spacemacs. I have also clarified that diff-hl is the preferred default tool for this task.

To avoid breaking configs I have changed the layer to treat the unsupported setting of git-gutter+ as diff-hl now.

If someone finds some time it would be great to migrate version control to spacemacs|defc vars to communicate miss configurations more clearly.
